lib/account.rb

```diff
@@ -0,0 +1,168 @@
+require 'csv'
+#require './lib/owner.rb'
+
+module Bank
+  class Account
+
+    attr_reader :balance, :id
+    attr_accessor :owner, :min_balance, :fee, :acct_type, :allowed_to_withdraw
+
+
+    def initialize(id, initial_balance, open_date)
+      @id = id
+      @balance = initial_balance.to_f.round(2)
+      @open_date = open_date
+      @min_balance =  0
+      @fee = 0
+      @owner = nil
+
+      # Raises an argument error if the initial balance is less than 0
+      if initial_balance.to_f < @min_balance
+        raise ArgumentError, "The balance cannot be less than 0."
+      end
+    end
+
+    # Creates accounts from the accounts.csv file
+    def self.all
+      @accounts = []
+      accounts_csv = CSV.read("support/accounts.csv")
+      # For each line in accounts pass the id, balance, and date (formatted as need be) into a new Bank Account object
+      accounts_csv.each do |id, balance, date|
+        balance = balance.to_f.round(2)/100
+        date = DateTime.strptime(date, "%Y-%m-%d %H:%M:%S %z")
+        id = Bank::Account.new(id, balance ,date)
```

Comment:
I would advise against using the `id` variable again here since it now stores an account variable rather than an identifier